src/components/scenes/CoinRankingDetailsScene.tsx

```diff
@@ -168,24 +172,87 @@ const CoinRankingDetailsSceneComponent = (props: Props) => {
     [edgeAssets, currencyWallets]
   )
 
-  /** Check if all the stake plugins are loaded for this asset type */
-  const isStakingLoading = matchingWallets.some(wallet => walletStakingStateMap[wallet.id] == null || walletStakingStateMap[wallet.id].isLoading)
+  /**
+   * Out of those wallets, which ones support staking, specific to the asset on
+   * this scene. This is only populated once the staking state has been
+   * initialized in the effect below.
+   */
+  const stakingWallets = matchingWallets.filter(wallet => {
+    return (
+      isStakingSupported(wallet.currencyInfo.pluginId, currencyCode) &&
+      walletStakingStateMap[wallet.id] != null &&
+      filterStakePolicies(
+        Object.values(walletStakingStateMap[wallet.id].stakePolicies).map(stakePolicy => stakePolicy),
+        { wallet, currencyCode }
+      ).length > 0
+    )
+  })
 
   React.useEffect(() => {
     if (isFocused && initFiat !== supportedFiat) {
       // Take this stale scene off the stack
       navigation.pop()
       // Force a refresh & refetch
       navigation.navigate('coinRanking')
+    }
+  }, [initFiat, isFocused, navigation, supportedFiat])
+
+  React.useEffect(() => {
+    // Initialize staking state
+    if (coinRankingData != null && matchingWallets.length > 0) {
+      // Start with a looser filter that does not include the stake policy,
+      // because it has not yet been initialized
+      const uninitializedStakingWallets = matchingWallets.filter(wallet => isStakingSupported(wallet.currencyInfo.pluginId, currencyCode))
+
+      uninitializedStakingWallets.forEach(wallet => {
+        if (walletStakingStateMap[wallet.id] != null && Object.keys(walletStakingStateMap[wallet.id].stakePolicies).length > 0) return
+        dispatch(updateStakingState(currencyCode, wallet)).catch(err => showError(err))
+      })
```

Comment:
Please swap `forEach` for `for (const wallet of uninitializedStakingWallets) { ...`, and of course the `return` becomes a `continue`. We hardly ever use `forEach` in our codebase, and it's slower anyhow.